src/ray/common/ray_syncer/ray_syncer.cc

```diff
@@ -320,8 +322,15 @@ void RaySyncer::Connect(std::unique_ptr<NodeSyncConnection> connection) {
 }
 
 void RaySyncer::Disconnect(const std::string &node_id) {
-  io_context_.post([this, node_id]() { sync_connections_.erase(node_id); },
-                   "RaySyncerDisconnect");
+  io_context_.post(
```

Comment:
Btw (unrelated to this PR), is there any reason why we have so many io context.post instead of making the class just thread safe?

Reply:
Yes, this is to ensure thread safe. Basically, for all public API, we need `io_context_.dispatch` (let me update post to dispatch)